tests/test_jetmet_tools.py

```diff
@@ -837,9 +837,9 @@ def test_corrected_jets_factory(optimization_enabled):
             **{name: evaluator[name] for name in jec_stack_names[5:6]}
         )
 
-        print(dak.necessary_columns(jets.eta))
+        print(dak.report_necessary_buffers(jets.eta))
```

Comment:
```suggestion
        print(dak.report_necessary_columns(jets.eta))
```

Reply:
is `dak.necessary_columns` deprecated? if so on what time scale?

Reply:
`necessary_columns` → `report_necessary_columns`.

We can make it as long as necessary. At one point, I was going to remove it in favour of `report_necessary_buffers`, but the new `report_necessary_columns` is designed to allow IO layers to report columns iff. they support them.